pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -93,9 +101,24 @@ func (p *producerJobResumer) Resume(ctx context.Context, execCtx interface{}) er
 		case <-p.timer.Ch():
 			p.timer.MarkRead()
 			p.timer.Reset(streamingccl.StreamReplicationStreamLivenessTrackFrequency.Get(execCfg.SV()))
-			j, err := execCfg.JobRegistry.LoadJob(ctx, p.job.ID())
+			var err error
+			var j *jobs.Job
+			for r := retry.Start(ro); r.Next(); {
+				j, err = execCfg.JobRegistry.LoadJob(ctx, p.job.ID())
+
+				if knobs := execCfg.StreamingTestingKnobs; knobs != nil && knobs.AfterResumerJobLoad != nil {
+					err = knobs.AfterResumerJobLoad(err)
+				}
+
+				// No need to retry if there is no job record.
+				if err == nil || jobs.HasJobNotFoundError(err) {
+					break
+				}
+
+				log.Warningf(ctx, "replication stream %d failed, may retry: %v", p.job.ID(), err)
+			}
 			if err != nil {
-				return err
+				return errors.Wrapf(err, "replication stream %d cannot resume", p.job.ID())
```

Comment:
Lets mention that this came from loading the job

Reply:
done.